git/git.go

```diff
@@ -71,17 +72,36 @@ func NewRepository(path string) (*Repository, error) {
 		gitBin: gitBin,
 	}
 
+	full, err := repo.IsFull()
+	if err != nil {
+		return nil, fmt.Errorf("determining whether the repository is a full clone: %w", err)
```

Comment:
Having worked on a partial clone just today, I misinterpreted the term "full clone" to mean "non-partial". So maybe "full" is a bit ambiguous and should be called "non-shallow" instead?

Reply:
That brings up a good question...should `git-sizer` be willing to operate on a partial clone?

The reason that it can't operate on a shallow clone is obvious...the objects that it needs are simply not there.

With a partial clone, I'm not certain what would happen. Would it fail because some objects are missing? Would those objects be pulled down "on demand"? If the latter, that would almost certainly be an unwelcome surprise. Either way, it should probably refuse to run, don't you think?

That's part of the reason that I called this method `IsFull()`, because I suspected that we might want to add some other criteria here.

Reply:
I'll leave this for now until we decide about partial clones. Oh and by the way, how does one detect that a repo is a partial clone?

Reply:
> The reason that it can't operate on a shallow clone is obvious...the objects that it needs are simply not there.

Well, from a naive point of view I would ask the question: why can't `git-sizer` work on the objects that are there, then, and ignore those that aren't there? The same would apply to partial clones, of course. It would make it possible to answer the question I hear a lot: "why on this burning planet is my local repository so large when I _specifically_ used a partial clone to prevent it from becoming so large?".